### DISCUSSION TITLE
Static type checker?

### DISCUSSION THREAD

**Discussion by domna** [PyEllips/pyElli#64] [Ideas]
Should we include a static type checker (e.g. [mypy](http://mypy-lang.org)) in the Actions?

**Comment by MarJMue** [comment]
I think we can do that, but as long as we use pretty generic types like npt.ArrayLike most of the time, it is of little use.
It would be much better if we would use something like [nptyping](https://github.com/ramonhagenaars/nptyping) to specify the shape of arrays.

**Comment by domna** [comment]
I see. So probably that's something we can implement in the future. However, I thought we could already include it if we work on the Actions for #62. Even if its not really helpful yet we could extend the typing in the future and have the checking already there. What do you think?

**Reply by MarJMue** [reply]
Sure lets include it. I think many error will still be functions not returning the proposed type, but unit test should cover that case. 

I quickly tried to run mypy and it seems we have to work a bit to make it work, because most of our dependencies are not typed and it does not like that :smile:

**Reply by domna** [reply]
I see :D Maybe we just have to tune the mypy to be less strict. Actually, I did not use mypy myself yet I just found it in one of our projects here. We're using the command `python -m mypy --ignore-missing-imports --follow-imports=silent --no-strict-optional <pkg>` in the actions and it is also mix-typed ;D. So maybe the arguments help with being less strict...
